Origin/moore garrett error controller

gmoore10 opened the pull request:
Added functionality to log errors to DB table. DB table was created using migrations.

For the next release, additional functionality to enable/disable presenting errors to an admin and fetching the last X number of errors will be implemented.

Commit:
Database Migration for ErrorLog Implented

Commit:
Implemented basic error logging

Still need to implement error fetching.

Commit:
Adjusted style guide tests to include bot app.

Kankur0 commented:
Everything works as intended for this specific release and passes the style guide tests.

We may want to add the two new commands to the help menu and only show it for Admins, but this may be something we have planned for the next release of this addition.

Kankur0 reviewed (approved):
Works as intended and passes Style Guide.

reactivepixel commented:
Very clean and well thought through approach. Solid work all around! Actual first try!
![First Try](https://i.imgur.com/4W6a7sg.gif)

Commit:
Implemented Error Toggling for Admins

Commit:
Fully Implemented toggleErrors and fetchErrors capabilities

Commit:
Added admin check for error messages

If toggleErrors is on but the user is no longer an admin, don't send the error message.

Kankur0 commented:
Great work on this, it's very in depth and helpful when looking into errors. I did however find two small things that we may want to change.

First on lines 109 and 110 of the `errors.js` file there are two util logs but no debug level. I'm assuming this was meant to either be removed or given a debug level. Maybe we should modify these two lines by giving them an appropriate level or removing them if they aren't needed.

The second thing is that doing `!fetchErrors <user>` gives me a "Complete" message but doesn't actually give any of the errors from that user.

Commit:
Fixed issue with parameters not being checked correctly.

Commit:
Bot kicks back DM on its own again

One of the commits broke the bot's ability to DM after being DM'd directly. (running a command in the DM instead of a channel). This issue was resolved.

Kankur0 reviewed (approved):
Errors properly get added to the database and admins are able to toggle alerts of the events on and off. Also allows admins to fetch all errors or by user with or without a limit. Great work!

Commit:
Refactored isServerAdmin() to behave like isElevatedRole()

Helps avoid violating DRY principle.

reactivepixel commented on bot/controllers/errors.js:
```diff
@@ -0,0 +1,161 @@
+    const { message } = this;
+    let params = [];
+
+    if (message.parsed[1] !== undefined) {
```
Magic Number

reactivepixel commented on bot/botUtils.js:
```diff
@@ -20,13 +20,19 @@ exports.generateCode = (n) => {
 // Checks if person is an admin user, use GuildMember object
-exports.isAdmin = (member) => {
-  const adminRoles = [
+exports.isInElevatedRole = (member) => {
```
A more well defined name for these users now, good call in making this change.
